trans/static/js/common.js

```diff
@@ -54,19 +54,52 @@ function sendPrintJob(print_task_url) {
 }
 
 function validateFinalizeTranslation(contest_slug, tasks_length) {
+    // Check tasks frozenness.
     const frozen_tasks = document.querySelectorAll(
         `.task-row[data-contest-slug="${contest_slug}"][data-task-frozen="True"]`
     );
-    if (frozen_tasks.length === tasks_length) {
-        return confirm('Are you sure?');
+    if (frozen_tasks.length !== tasks_length) {
+        const unfrozen_tasks = document.querySelectorAll(
+            `.task-row[data-contest-slug="${contest_slug}"][data-task-frozen="False"]`
+        );
+        const unfrozen_task_names = Array.from(unfrozen_tasks)
+              .map((el) => { return el.dataset.taskName});
+        alert(`Please freeze the following task(s) first: ${unfrozen_task_names}`);
+        return false;
     }
 
-    const unfrozen_tasks = document.querySelectorAll(
-        `.task-row[data-contest-slug="${contest_slug}"][data-task-frozen="False"]`
-    );
-    const unfrozen_task_names = Array.from(unfrozen_tasks)
-          .map((el) => { return el.dataset.taskName});
-    alert(`Please freeze the following task(s) first: ${unfrozen_task_names}`);
-    return false;
-}
+    // It doesn't make sense for the two extra countries to be the same.
+    const querySelectElement = (name) =>
+          document.querySelector(
+              `form[data-contest-slug="${contest_slug}"] select[name="${name}"]`
+          ).value;
+    const extra_country_1_code = querySelectElement('extra_country_1_code');
+    const extra_country_1_count = parseInt(querySelectElement('extra_country_1_count'));
+    const extra_country_2_code = querySelectElement('extra_country_2_code');
+    const extra_country_2_count = parseInt(querySelectElement('extra_country_2_count'));
+    if (extra_country_1_count > 0 && extra_country_2_count > 0 &&
+        extra_country_1_code === extra_country_2_code) {
+        alert('Additional copy #1 & #2 should be different!');
+        return false;
+    }
 
+    // Build confirmation message.
+    const confirmation_messages = [];
+    if (extra_country_1_code.length > 0) {
+        let detail = `${extra_country_1_code}: ${extra_country_1_count}`;
+        if (extra_country_1_count === 0) {
+            detail += '\n  Did you forget to specify the number of copies?';
+        }
+        confirmation_messages.push(detail);
```

Comment:
I think instead confirming, in this case we need to alert & return false, because it does not really make sense to have extra country with 0 copies.

Reply:
Done.